app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  before_save :downcase_email
+
+  validates :email, presence: true, length: {minium: 20, maximum: 255},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+
+  validates :password, presence: true, length: {minimum: 6}, if: :password
+  has_secure_password
+  private
+
+  def downcase_email
+    email.downcase!
```

Comment:
downcase vs downcase! khác nhau như thế nào

Reply:
downcase trả về một chuỗi mới in thường
downcase! thay đổi trực tiếp chuỗi hiện tại thành in thường
downcase! sẽ trả về nil nếu như xâu ký tự không có thay đổi